pkg/image/api/v1/conversion.go

```diff
@@ -97,13 +95,13 @@ func convert_v1_Image_To_api_Image(in *Image, out *newer.Image, s conversion.Sco
 	return nil
 }
 
-func convert_v1_ImageStreamSpec_To_api_ImageStreamSpec(in *ImageStreamSpec, out *newer.ImageStreamSpec, s conversion.Scope) error {
+func Convert_v1_ImageStreamSpec_To_api_ImageStreamSpec(in *ImageStreamSpec, out *newer.ImageStreamSpec, s conversion.Scope) error {
 	out.DockerImageRepository = in.DockerImageRepository
 	out.Tags = make(map[string]newer.TagReference)
 	return s.Convert(&in.Tags, &out.Tags, 0)
 }
 
-func convert_api_ImageStreamSpec_To_v1_ImageStreamSpec(in *newer.ImageStreamSpec, out *ImageStreamSpec, s conversion.Scope) error {
+func Convert_api_ImageStreamSpec_To_v1_ImageStreamSpec(in *newer.ImageStreamSpec, out *ImageStreamSpec, s conversion.Scope) error {
```

Comment:
You're not crazy.  I'm not crazy about hand generated functions anymore
though.

On Wed, Mar 9, 2016 at 7:54 AM, David Eads <notifications@github.com> wrote:

> In pkg/image/api/v1/conversion.go
> <https://github.com/openshift/origin/pull/7814#discussion_r55513277>:
>
> >  	out.DockerImageRepository = in.DockerImageRepository
> >  	out.Tags = make(map[string]newer.TagReference)
> >  	return s.Convert(&in.Tags, &out.Tags, 0)
> >  }
> >
> > -func convert_api_ImageStreamSpec_To_v1_ImageStreamSpec(in *newer.ImageStreamSpec, out *ImageStreamSpec, s conversion.Scope) error {
> > +func Convert_api_ImageStreamSpec_To_v1_ImageStreamSpec(in *newer.ImageStreamSpec, out *ImageStreamSpec, s conversion.Scope) error {
>
> This is the only method in image that has defaulting
>
> If we add defaulting, we'd have to come in here and remember to update the
> associated conversion function. I'm not convinced that we'd always get that
> right. Unit tests will miss it because they usually test the defaulter
> directly, fuzzer tests tend to specifically exclude defaulting scenarios,
> and integration/e2e tests aren't written for defaulting.
>
> I know we're worried about reflection calls, but code-rot is also a real
> problem.
>
> @liggitt <https://github.com/liggitt> Am I crazy to be worried about this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7814/files#r55513277>.
>